source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -1508,7 +1508,7 @@ void ShaderGraph::calculateScopes()
                 {
                     newScopeInfo.adjustAtConditionalInput(node, int(inputIndex), 0x12);
                 }
-                else if (isSwitch)
+                else if (isSwitch && inputIndex != node->numInputs() - 1)
```

Comment:
Not mine. Possibly broke when the `which` "parameter" became an "input" with 1.38.